File: database/redis/__demo__/bloom-filter-demo/src/main/java/xyz/icefery/demo/util/R.java
package xyz.icefery.demo.util;

public record R<T>(
    String code,
    String message,
    T data
) {
    public static R<Void> success() {
        return new R<>("0", "success", null);
    }

    public static <T> R<T> success(T data) {
        return new R<>("0", "success", data);
    }

    public static R<Void> failure() {
        return new R<>("-1", "failure", null);
    }

    public static R<Void> failure(String code, String message) {
        return new R<>(code, message, null);
    }
}

File: root/lang/rust/__demo__/rocket-demo/src/utils/r.rs
#[derive(serde::Serialize, serde::Deserialize)]
pub struct R<T> {
    code: i32,
    message: String,
    data: Option<T>,
}

impl<T> R<T> {
    pub fn success(data: T) -> Self {
        Self {
            code: 0,
            message: "success".to_string(),
            data: Some(data),
        }
    }

    pub fn failure(message: String) -> Self {
        Self { code: -1, message, data: None }
    }
}
